addons/web/static/src/core/file_viewer/file_model.js

```diff
@@ -130,6 +130,9 @@ export const FileModelMixin = (T) =>
             if (this.uploading && this.tmpUrl) {
                 return this.tmpUrl;
             }
+            if (!this.id) {
+                return this.url;
```

Comment:
https://github.com/odoo/odoo/pull/197539/files#diff-1e34ac32dfe87be0246aa68e6902baf0629887c86ef7f4013784509f5fb7dd38R19

@seb-odoo could you review this PR?

Reply:
Hello @seb-odoo, thanks for taking the time to look into this.

Regarding the test:  
From my investigation, I found that `FileModelMixin` is used in only two places within the codebase: it's extended by both `Attachment` and `FileModel`. For `Attachment`, it makes sense to always expect an ID. However, for `FileModel`, it appears to have been introduced specifically in the context of changes within the Knowledge app, and this is the only place it's currently used.

Given this, I decide to include the test within the Knowledge app (see the related Enterprise PR) rather than in the Web module, mainly because it seemed more relevant and, honestly, easier. 😄

Let me know if you would prefer to have it in web.

Reply:
This also highlights the problem I have with this mixin, it's way too specific and obscure.

I guess it's fine if the test is in knowledge for now (at least to test the knowledge feature is working fine).

It would be very easy to add a test in web too, as simple as "create `FileModel` instance -> set url and no id -> check result of urlRoute".

Reply:
I just added the test 👍🏾

Reply:
An important part would be to check attachment with id + url at the same time prioritizes the id (this was the fix of the last PR changing this file, we have a functional test to cover it in mail, but it might be worth adding it here too).

Other than that, I let @aab-odoo do the final check.

Reply:
Good catch. Done!